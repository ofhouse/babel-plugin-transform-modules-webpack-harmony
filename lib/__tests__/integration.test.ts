import { transform as babelTransform } from '@babel/core';
import { wrap } from 'jest-snapshot-serializer-raw';
import reactPreset from '@babel/preset-react';

import plugin from '../babel-plugin-transform-modules-webpack-harmony';

function transform(input, options = {}) {
  return wrap(
    babelTransform(input, {
      babelrc: false,
      configFile: false,
      presets: [reactPreset],
      plugins: [[plugin, options]],
    }).code
  );
}

describe('plugin-transform-modules-webpack-harmony', () => {
  test('JSX code', () => {
    expect(
      transform(`
      import React from "react";
      import { Blocks, HeaderLogo, HeaderBasic } from "@blocks/react";
      export default () => (
        <Blocks.Root>
          <HeaderLogo>
            <HeaderLogo.Logo to="/" />
            <HeaderLogo.Nav>
              <HeaderLogo.Link to="/about">About</HeaderLogo.Link>
              <HeaderLogo.Link to="/blog">Blog</HeaderLogo.Link>
              <HeaderLogo.Link to="/contact">Contact</HeaderLogo.Link>
            </HeaderLogo.Nav>
          </HeaderLogo>
          <HeaderBasic>
            <HeaderBasic.Logo to="/">Hello</HeaderBasic.Logo>
            <HeaderBasic.Nav>
              <HeaderBasic.Link to="/about">About</HeaderBasic.Link>
              <HeaderBasic.Link to="/blog">Blog</HeaderBasic.Link>
              <HeaderBasic.Link to="/contact">Contact</HeaderBasic.Link>
            </HeaderBasic.Nav>
          </HeaderBasic>
        </Blocks.Root>
      );
    `)
    ).toMatchSnapshot();
  });
});
