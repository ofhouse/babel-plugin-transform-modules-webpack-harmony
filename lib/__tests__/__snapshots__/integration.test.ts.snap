// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin-transform-modules-webpack-harmony JSX code 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): "var react__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(\\"node_modules/react\\");

var react__WEBPACK_IMPORTED_MODULE_0___default = __webpack_require__.n(react__WEBPACK_IMPORTED_MODULE_0__);

var _blocks_react__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(\\"node_modules/@blocks/react\\");

var _ref = () => react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"Blocks\\"].Root, null, react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"], null, react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"].Logo, {
  to: \\"/\\"
}), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"].Nav, null, react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"].Link, {
  to: \\"/about\\"
}, \\"About\\"), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"].Link, {
  to: \\"/blog\\"
}, \\"Blog\\"), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderLogo\\"].Link, {
  to: \\"/contact\\"
}, \\"Contact\\"))), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"], null, react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"].Logo, {
  to: \\"/\\"
}, \\"Hello\\"), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"].Nav, null, react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"].Link, {
  to: \\"/about\\"
}, \\"About\\"), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"].Link, {
  to: \\"/blog\\"
}, \\"Blog\\"), react__WEBPACK_IMPORTED_MODULE_0__.createElement(_blocks_react__WEBPACK_IMPORTED_MODULE_1__[\\"HeaderBasic\\"].Link, {
  to: \\"/contact\\"
}, \\"Contact\\"))));

__webpack_exports__[\\"default\\"] = (_ref);",
}
`;
